src/messages/manifestjson.js

```diff
@@ -22,13 +22,39 @@ export const MANIFEST_FIELD_INVALID = {
 };
 
 export const MANIFEST_FIELD_UNSUPPORTED = 'MANIFEST_FIELD_UNSUPPORTED';
-export function manifestFieldUnsupported(fieldName) {
+export function manifestFieldUnsupported(fieldName, error) {
+  const versionRange = errorParamsToUnsupportedVersionRange(error.params);
+  const messageTmpl = versionRange
+    ? i18n._(oneLine`"%(fieldName)s" is in a format not supported in
+                     manifest versions %(versionRange)s.`)
+    : i18n._(oneLine`"%(fieldName)s" is in an unsupported format.`);
+  const message = i18n.sprintf(messageTmpl, { fieldName, versionRange });
+
   return {
-    code: 'MANIFEST_FIELD_UNSUPPORTED',
-    message: i18n._('Manifest field not supported.'),
-    description: i18n.sprintf(i18n._('"%(fieldName)s" is not supported.'), {
-      fieldName,
-    }),
+    code: MANIFEST_FIELD_UNSUPPORTED,
+    message,
+    description: message,
+    file: MANIFEST_JSON,
+  };
+}
+
+export const MANIFEST_PERM_UNSUPPORTED = 'MANIFEST_PERM_UNSUPPORTED';
+export function manifestPermUnsupported(permissionName, error) {
+  const versionRange = errorParamsToUnsupportedVersionRange(error.params);
+  const messageTmpl = versionRange
+    ? i18n._(oneLine`/%(fieldName)s: "%(permissionName)s" is not supported in
```

Comment:
Why is the forward slash needed in those messages?

Reply:
@willdurand honestly I just kept using the same convention used elsewhere, the logic seems to be that `fieldName` is the path to the manifest property that the error refers to, and so as an example:

- if the error is related to the background scripts in the manifest, in the error we translate that into `"/background/scripts"`
- and if the error is related to a permission, in the error we translate that into `"/permissions"` or `"/optional_permissions"`

Reply:
Ok, that works for me. I am not too familar with the linter output tbh..